test/e2e/downwardapi_volume.go

```diff
@@ -18,57 +18,180 @@ package e2e
+var podLogTimeout = 15 * time.Second
+
 var _ = Describe("Downward API volume", func() {
 	f := NewFramework("downward-api")
-
-	It("should provide labels and annotations files [Conformance]", func() {
+	It("should provide podname only [Conformance]", func() {
 		podName := "downwardapi-volume-" + string(util.NewUUID())
-		pod := downwardAPIVolumePod(podName)
+		pod := downwardAPIVolumePod(podName, map[string]string{}, map[string]string{}, "/etc/podname")
 
 		testContainerOutput("downward API volume plugin", f.Client, pod, 0, []string{
-			fmt.Sprintf("cluster=\"rack10\"\n"),
-			fmt.Sprintf("builder=\"john-doe\"\n"),
 			fmt.Sprintf("%s\n", podName),
 		}, f.Namespace.Name)
 	})
 
-	It("should provide labels and annotations files as non-root with fsgroup [Conformance] [Skipped]", func() {
+	It("should provide podname as non-root with fsgroup [Conformance] [Skipped]", func() {
 		podName := "metadata-volume-" + string(util.NewUUID())
 		uid := int64(1001)
 		gid := int64(1234)
-		pod := downwardAPIVolumePod(podName)
+		pod := downwardAPIVolumePod(podName, map[string]string{}, map[string]string{}, "/etc/podname")
 		pod.Spec.SecurityContext = &api.PodSecurityContext{
 			RunAsUser: &uid,
 			FSGroup:   &gid,
 		}
 		testContainerOutput("downward API volume plugin", f.Client, pod, 0, []string{
-			fmt.Sprintf("cluster=\"rack10\"\n"),
-			fmt.Sprintf("builder=\"john-doe\"\n"),
 			fmt.Sprintf("%s\n", podName),
 		}, f.Namespace.Name)
 	})
+
+	It("should verify update on labels modification [Conformance]", func() {
+		labels := map[string]string{}
+		labels["key1"] = "value1"
+		labels["key2"] = "value2"
+
+		podName := "labelsupdate" + string(util.NewUUID())
+		pod := downwardAPIVolumePod(podName, labels, map[string]string{}, "/etc/labels")
+		containerName := "client-container"
+		defer func() {
+			By("Deleting the pod")
+			f.Client.Pods(f.Namespace.Name).Delete(pod.Name, api.NewDeleteOptions(0))
+		}()
+		By("Creating the pod")
+		_, err := f.Client.Pods(f.Namespace.Name).Create(pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		expectNoError(waitForPodRunningInNamespace(f.Client, pod.Name, f.Namespace.Name))
+
+		pod, err = f.Client.Pods(f.Namespace.Name).Get(pod.Name)
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() string {
+			logs, err := f.Client.Get().
+				Resource("pods").
+				Namespace(f.Namespace.Name).
+				Name(pod.Name).
+				SubResource("log").
+				Param("container", containerName).
+				Do().
+				Raw()
+			if err == nil && strings.Contains(string(logs), "Internal Error") || err != nil {
+				By(fmt.Sprintf("Error retreving logs %s/%s/%s", f.Namespace.Name, podName, containerName))
+				return ""
+			}
+			return string(logs)
+		},
+			podLogTimeout, poll).Should(ContainSubstring("key1=\"value1\"\n"))
+
+		//modify labels
+		pod.Labels["key3"] = "value3"
+		_, err = f.Client.Pods(f.Namespace.Name).Update(pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() string {
+			logs, err := f.Client.Get().
```

Comment:
this is what I meant when I was mentioning to @pmorie  that the `Eventually` is a little bit more [verbose[(https://github.com/kubernetes/kubernetes/pull/15502#discussion_r45320029). Unluckily `Eventually` takes as input only a `func () ...` no input parameters so I've to inline the `func` [see](https://onsi.github.io/gomega/#making-asynchronous-assertions). As I was answerging to @pmorie I'm ok to modify this (basically adding the ability to skip pod creation to `testContainerOutput`), [see](https://github.com/kubernetes/kubernetes/pull/15502#discussion_r45322004).
@ixdy not sure if this is acceptable for you.